### DISCUSSION TITLE
Change floating-point .str() result

### DISCUSSION THREAD

**Discussion by benstigsen** [vlang/v#13720] [Ideas]
Let's say we have something like this:
```v
a := 3.0
b := 4.4
println('$a + $b = ${a+b}')
```
**What I would expect:**
`3.0 + 4.4 = 7.4`
or
`3 + 4.4 = 7.4`

**What the result actually is:**
`3. + 4.4 = 7.4`

---

I think that this is not a good way to do it. I get that by printing the decimal point, it is shown the number is a float, but it looks weird to just end with a dot. I'd much rather prefer that it shows just `3`, or if it has to be shown that it's a float, I'd prefer that it shows `3.0`.

**Comment by vincenzopalazzo** [comment]
I agree with you, I aspect that the str method print the number in the correct format, so I would like a result `3.0` and not just `3.`

**Comment by dumblob** [comment]
First of all, `println()` and friends seem still to not behave as expected (read "they're buggy"): https://github.com/vlang/v/discussions/11855#discussioncomment-1366384

Second, the syntax above `println($a + $b = ${a+b})` is obviously wrong - you probably meant `println( '$a + $b = ${a+b}' )`.

Third, the code outputs `3 + 4.4 = 7.4` (not `3. + 4.4 = 7.4` as in your example) with the latest *V* on Linux. Are you also on Linux or something else? Could you try with the latest *V* build?

**Reply by vincenzopalazzo** [reply]
can you open an issue with a minimal example? looks that you had it already

**Reply by dumblob** [reply]
Before opening a new issue I'd like to have the **existing** issues solved :wink:. I've now asked for reopening of https://github.com/vlang/v/issues/9915 in https://github.com/vlang/v/discussions/11855#discussioncomment-2510869 .

I've now also added new comments to the already-linked PR https://github.com/vlang/v/issues/7692 .

If you still think we should open separate issues for all these things, let me know.

**Reply by benstigsen** [reply]
You are correct in that I forgot to put quotes.
But, I am 1000% certain that it printed `3.` before. I do see now that this is not the case, but that's probably due to it having been changed within the last month or so.

**Reply by benstigsen** [reply]
Actually, doing:
```v
a := 3.0
println(a)
```

Results in: `3.`, so it is most like the string interpolation/formatting that has been changed.

**Reply by dumblob** [reply]
Yes, that's one of the issues of `println()` (and friends) being special :cry: - it's a known long-standing issue. IDK if anyone is working on correcting all these bugs in `println()` and friends.

**Reply by benstigsen** [reply]
I have a slight feeling that `println()` or string interpolation in general may be overly complicated. It might be better to just assume that each variable or value in the string formatting has a `.str()` method on it. Making use of that, instead of doing conversions in the string interpolation code it self, since the code is quite _big_ to say the least. This would also make it so that instead of printing the value different when doing:
```v
a := 3.0
println(a) // 3.
println('$a') // 3
```
Both would result in the same if they just try to call the `a.str()` method instead.
